lib/rules/no-constant-condition.js

```diff
@@ -97,7 +97,8 @@ module.exports = {
                 case "ObjectExpression":
                 case "ArrayExpression":
                     return true;
-
+                case "TemplateLiteral":
+                    return node.expressions.every(isConstant);
```

Comment:
`Array.prototype.every` will also pass element's index as the second argument to `isConstant`.

This will become `inBooleanPosition` and produce some inconsistency which is observable with `typeof`:

```js
if (`${typeof a}`) {} // no error because of falsy 0

if (`${''}${typeof a}`) {} // error because of truthy 1
```